File: JJB/BaekJoon/231209/10828_스택.py
import sys

N = int(sys.stdin.readline())

s = []

for i in range(N):
    line = sys.stdin.readline().split()
    command = line[0]
    if command == "push":
        number = int(line[1])

    if command == "push":
        s.append(number)
    elif command == "pop":
        if not s:
            print(-1)
        else:
            print(s[-1])
            s.pop()
    elif command == "size":
        print(len(s))
    elif command == "empty":
        print(int(not s))
    elif command == "top":
        if not s:
            print(-1)
        else:
            print(s[-1])

File: JJB/BaekJoon/231209/10828_스택.cpp
#include <iostream>
#include <string>
#include <stack>

using namespace std;

int main()
{
	int N;
	cin >> N;

	stack<int> s;
	
	for (int i = 0; i < N; i++)
	{
		string command;
		cin >> command;
		int number;
		if (command == "push")
			cin >> number;

		if (command == "push")
			s.push(number);
		else if (command == "pop")
		{
			if (s.empty())
				cout << -1 << endl;
			else
			{
				cout << s.top() << endl;;
				s.pop();
			}
		}
		else if (command == "size")
			cout << s.size() << endl;
		else if (command == "empty")
			cout << s.empty() << endl;
		else if (command == "top")
		{
			if (s.empty())
				cout << -1 << endl;
			else
				cout << s.top() << endl;
		}
	}
}